src/components/loaders/useLoaderAlerts.js

```diff
@@ -0,0 +1,87 @@
+import { useAlert } from '@dhis2/app-service-alerts'
+import i18n from '@dhis2/d2-i18n'
+import {
+    ALERT_MESSAGE_DYNAMIC,
+    INFO_NO_DATA,
+    WARNING_NO_OU_COORD,
+    WARNING_NO_GEOMETRY_COORD,
+    ERROR_CRITICAL,
+    CUSTOM_ALERT,
+} from '../../constants/alerts.js'
+
+function useLoaderAlerts(loaderAlertAction) {
+    const errorAlert = useAlert(ALERT_MESSAGE_DYNAMIC, {
+        critical: true,
+        onHidden: loaderAlertAction,
+    })
+    const warningAlert = useAlert(ALERT_MESSAGE_DYNAMIC, {
+        warning: true,
+        onHidden: loaderAlertAction,
+    })
+    const infoAlert = useAlert(ALERT_MESSAGE_DYNAMIC, {
+        info: true,
+        duration: 5000,
+        onHidden: loaderAlertAction,
+    })
+    const noDataAlert = useAlert(ALERT_MESSAGE_DYNAMIC, {
+        info: true,
+        duration: 5000,
```

Comment:
It seems a bit inconsistant to me that no geometry is getting the warning treatment and is persisting while no data is only an info and disappears.

Reply:
I can definitely reconsider that. Currently "no data" is a warning, but that is not in line with our other apps, which simply present the empty chart/table/linelist, and "no data" not necessarily meaning there is something wrong. That's why I changed it. One improvement could be to add a "no data" indicator to the Layer Card? That could be implemented later once Joe is back.

You raise a good question as to whether "no geometry" and "no data" should behave the same. To me, they are slightly different. I was thinking of geometry as more of an instance configuration, the user was clearly expecting those to be defined and they weren't.

I could switch "no data" back to a warning, to be consistent with current behavior. Then we could address the way we do alerts altogether, once Joe is back.